### PR TITLE
Add keysFile config validations, defaults, and tests

### PR DESCRIPTION
**Pull request by davnicwil**
All tests passing, although I do see this in the output

image

Not sure if that is supposed to be intentional?

### CODE REVIEW

**Comment by davnicwil** [comment]
Ah ok I assume that is this `it('fails to retrieve localizations', async () => {`

**Comment by carlossless** [comment]
Yeah, jest will pipe through anything streamed into stderr or stdout, and also will display caught exceptions.

File: src/config.test.js [hunk 1]
```diff
@@ -8,21 +8,68 @@ describe('config', () => {
...
+      expect(conf1).toEqual({
         token: 'test_token',
         project: 'test_project',
         output: 'test_output'
       })
+
+      // with LOKALISE_KEYS_FILE set true, should be all default settings
+      process.env.LOKALISE_KEYS_FILE = 'true'
+
+      const conf2 = await config.build(null, { })
```
Comment (carlossless) [inline src/config.test.js]: these cases could be different specs nested under a single context.

**Comment by carlossless** [comment]
I'm closing this as I ported over and covered the bulk of this functionality.

**Pull request closed without merging**